src/internal.rs

```diff
@@ -505,27 +507,49 @@ pub struct Rav1dFrameContext_frame_thread {
 #[derive(Default)]
 pub(crate) struct TxLpfRightEdge {
     /// `.len() = h * 2`
-    inner: Vec<u8>,
+    inner: DisjointMut<Vec<u8>>,
 }
 
 impl TxLpfRightEdge {
     #[allow(dead_code)]
-    pub const fn new() -> Self {
-        Self { inner: Vec::new() }
+    pub fn new() -> Self {
+        Self {
+            inner: DisjointMut::default(),
+        }
     }
 
     pub fn resize(&mut self, right_edge_size: usize, value: u8) {
         self.inner.resize(right_edge_size * 32 * 2, value)
     }
 
-    pub fn get(&self) -> (&[u8], &[u8]) {
+    pub fn get<'a>(
+        &'a self,
+        index_y: Range<usize>,
+        index_uv: Range<usize>,
+    ) -> (
+        impl 'a + Deref<Target = [u8]>,
+        impl 'a + Deref<Target = [u8]>,
+    ) {
         let mid = self.inner.len() / 2;
-        self.inner.split_at(mid)
+        assert!(index_y.end <= mid);
+        let (uv_start, uv_end) = (index_uv.start + mid, index_uv.end + mid);
+        (
+            self.inner.index(index_y),
+            self.inner.index(uv_start..uv_end),
+        )
+    }
+
+    pub fn copy_from_slice_y(&self, index: Range<usize>, src: &[u8]) {
```

Comment:
Don't all of these methods, `fn copy_from_slice_{y,uv}` and `fn get`, need to be `unsafe`?  Their safeness relies on the specific `index`es passed.  It's clear that `y` and `uv` are non-overlapping (statically known), but within `y` and `uv`, the `index`es could overlap, so these all need to be `unsafe`, right?